### ISSUE TITLE
Table - CSV export to Excel encoding issue

### ISSUE THREAD

**Issue by anthraxn8b** [grafana/grafana#32516]
**What happened**:
Special charaters are not converted correctly.
Having a table with transformed headlines (maybe also without) containing special characters (like the german äöü) are not converted correctly.
Opening the CSV with Excel (exported with Excel dialect) show strange signs instead of the correct special chars.

**What you expected to happen**:
Correctly encoded chars.

**How to reproduce it (as minimally and precisely as possible)**:
Look at the problem description.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.5.1
- Data source type & version: SimpleJSON
- OS Grafana is installed on: Windows 2016 Server
- User OS & Browser: Wundows/Macos Chrome
- Grafana plugins:
- Others:

**Comment by Elfo404** [comment]
Can't reproduce this as i don't have access to Excel, but I can see the correct values in google docs

**Comment by leeoniya** [comment]
what i know about Excel and how it treats opening non-ascii CSVs (like UTF-8) is that they must have a BOM mark to open properly. there's a good chance it's not being inserted during export.

tackled this last year in another repo: https://github.com/HeidiSQL/HeidiSQL/issues/1077

**Comment by anthraxn8b** [comment]
I will add some more sugar on tuesday.

**Comment by Blaimi** [comment]
exported CSVs have a BOM since 7.1, see #24924 / #26145.

**Comment by Blaimi** [comment]
We made an option to change the delimiter to something else (https://github.com/Blaimi/grafana/tree/csv_delimiter_option) also with some configurable defaults via grafana.ini. The commit for the default-value has a bug but we are on it.

This is not a solution but only a workaround. As long as Microsoft doesn't fix this in Excel, IMHO changing the delimiter to the one your Excel-locale expects is the best solution.

**Comment by jlaudenbach** [comment]
Thank you for the fast answer.

In meanwhile I also understand that the issue is on Microsoft side. The expression `sep=` just kills the UTF8 conversion. 
And the Problem was also there in Grafana version `6.2.5`. But there it was possible to get proper CSV-File when not using `Excel CSV Dialect`. 
For completion (and/or for other readers) here is what I figured for CSV export between Grafana version `6.2.5` and `8.4.3`

----
Grafana version  `6.2.5`:
---
Exported CSV without option `Excel CSV Dialect` will be like:
 ```
 "header_1";"header_2";"header_3"
 "März";"other";"2"
 ```
 Excel auto-import will:
* Separate the different columns: Yes
* Keep special chars: Yes

Exported CSV with option `Excel CSV Dialect` will be like:
```
sep=;
"header_1";"header_2";"header_3"
"März";"other";"2"
```
 Excel auto-import will:
* Separate the different columns: Yes
* Keep special chars: No

----
Grafana version  `8.2.5`:
---
Exported CSV withoption `Download for Excel` will be like:
```
"header_1","header_2","header_3"
März,other,2
```
Excel auto-import will:
* Separate the different columns: No
* Keep special chars: Yes

Exported CSV with option `Download for Excel` will be like:
```
sep=,
"header_1";"header_2";"header_3"
März,other,2
```
Excel auto-import will:
* Separate the different columns: Yes
* Keep special chars: No


In summary: In my case using Grafana version `6.2.5` without option `Excel CSV Dialect` gave a useful CSV to Excel-autoimport. And it seems that `""` are missing in Grafana Version `8.2.5` for column entries.

**Comment by jlaudenbach** [comment]
I think the problem is still an issue, or is there meanwhile a option to set the default delimiter from `,` to `;` for excel-dialect data export?
I couldn't found in Grafana 11 settings. 

Best Greetings

**Comment by Blaimi** [comment]
I created this option in a fork: https://github.com/grafana/grafana/commit/79c5972f7675ad6d09cb28d938f22da924c709a8 (as part of https://github.com/grafana/grafana/compare/v10.2.x...Blaimi:grafana:v10.2.x). You can even set the default delimiter per instance in `grafana.ini` (https://github.com/grafana/grafana/commit/068c607f23debea6372a0e54dbb5b83b003c481a) and in the user-settings (https://github.com/grafana/grafana/commit/452b3b48a9a61e4e08ddd269b82eb3a2464f92f6). Caution: The code is ugly and has no tests. Also missing: Configuration for the decimal separator (`,` vs `.` – same problem here)

**Comment by jlaudenbach** [comment]
Hi @Blaimi,
thank you for your reply, but unfortunately I can not use a fork. 
Do you know if the issue will  reopened automatically after we add new posts to it?

**Comment by samjewell** [comment]
It didn't seem to be reopened automatically, but I figured I should reopen it for you, given that it seems this is still an issue, and that it was closed only because it went stale.

**Comment by jlaudenbach** [comment]
Thank you @samjewell for reopening the issue.

**Comment by hfp-robloh** [comment]
It would be handy if we could specify delimiters to be used when exporting to Excel, i.e. not through `sep=` lines but in the rows directly.

For example, if we know our users are expecting semicolons (as UTF-8 dependents sometimes do...), then we can export with semicolons separating the values. E.g:

```
"Col1";"Col2"
Löwe;24
```

Then these semicolon-delimiting Users can simply double-click the file. It will load with the UTF-8 interpretation specified by the BOM intact.

**Comment by realhuseyn** [comment]
any updates?

**Comment by esprit64** [comment]
Grafana version: 11.3.1
Special charaters are not converted correctly.
Having a table with transformed headlines (maybe also without) containing special characters (like the Persian ø§üÿ) are not converted correctly.
Opening the CSV with Excel (exported with Excel dialect) show strange signs instead of the correct special chars.
this issue is resovled after downloading and editing the csv, but it is better to resolve the export issue of grafana ....

**Comment by lexa322** [comment]
Grafana version: 12.1.0
The encoding issues persist.
Is there a timeframe for a solution?

Does anyone have a lifehack?

**Comment by jlaudenbach** [comment]
For me it works if you do **not** use the _Download for Excel_ option for CSV download. 
And than open the file in Excel with CSV import, not just double click on the file.
